app/src/main/java/fr/free/nrw/commons/upload/UploadMediaDetail.java

```diff
@@ -1,17 +1,49 @@
 package fr.free.nrw.commons.upload;
 
+import android.util.Log;
+
 import java.util.List;
 
+import timber.log.Timber;
+
 /**
  * Holds a description of an item being uploaded by {@link UploadActivity}
  */
-public class Description {
+public class UploadMediaDetail {
 
     private String languageCode;
     private String descriptionText;
+    public String captionText;
     private int selectedLanguageIndex = -1;
     private boolean isManuallyAdded=false;
 
+    /**
+     * Formatting captions to the Wikibase format for sending labels
+     * @param uploadMediaDetails list of media Details*/
+
+    public static String formatCaptions(List<UploadMediaDetail> uploadMediaDetails) {
+        StringBuilder captionListString = new StringBuilder();
+        captionListString.append("{\\\"labels\\\":{");
```

Comment:
This is very confusing and prone to error, Please use JsonObject and obect.toJson()(using Gson), for maing this JSONObject